syntax="proto3";

service KVStore {
    rpc Put(PutRequest) returns (PutResponse) {}
    rpc Get(GetRequest) returns (GetResponse) {}
}

message PutRequest {
    string key=1;
    string value=2;
}

message PutResponse {
    string error=1;
}

message GetRequest {
    string key=1;
}

message GetResponse {
    string value=1;
    bool hit=2;
    string error=3;
}